File: BackEnd/src/domain/interfaces/usecases/plants/IUpdatePlantUseCase.ts
/**
 * @file IUpdatePlantUseCase.ts
 * @description Interfaz del caso de uso para actualizar una planta.
 * @module Plants
 * @layer Domain
 */

import type { UpdatePlantRequestDTO } from '../../../dtos/plants/update-plant-request.dto.js';
import type { PlantResponseDTO } from '../../../dtos/plants/plant-response.dto.js';

export interface IUpdatePlantUseCase {
  /**
   * @param plantId — Id de la planta a actualizar.
   * @param dto — Campos a actualizar (todos opcionales).
   * @param userId — Id del usuario (para verificar ownership).
   * @returns Planta actualizada como DTO de respuesta.
   * @throws {NotFoundException} Si la planta no existe.
   * @throws {UnauthorizedException} Si el usuario no es propietario.
   */
  execute(plantId: string, dto: UpdatePlantRequestDTO, userId: string): Promise<PlantResponseDTO>;
}

File: FrontEnd/lib/domain/interfaces/usecases/plants/i_update_plant_use_case.dart
/// @file i_update_plant_use_case.dart
/// @description Interfaz del caso de uso para actualizar una planta.
/// @module Plants
/// @layer Domain
library;

import '../../../dtos/plants/update_plant_request_dto.dart';
import '../../../entities/plant.dart';

abstract interface class IUpdatePlantUseCase {
  /// Actualiza la planta con [plantId] con los datos de [dto].
  Future<Plant> execute(String plantId, UpdatePlantRequestDto dto);
}
